QModManager/Patching/Patcher.cs

```diff
@@ -0,0 +1,171 @@
+namespace QModManager.Patching
+{
+    using System;
+    using System.IO;
+    using System.Reflection;
+    using System.Text.RegularExpressions;
+    using API;
+    using API.ModLoading;
+    using API.ModLoading.Internal;
+    using Checks;
+    using Harmony;
+    using QModManager.DataStructures;
+    using Utility;
+
+    /// <summary>
+    /// The main class which handles all of QModManager's patching
+    /// </summary>
+    public static class Patcher
+    {
+        internal static HarmonyInstance Harmony;
+
+        internal static readonly Regex IDRegex = new Regex("[^0-9a-z_]", RegexOptions.IgnoreCase);
+
+        internal static string QModBaseDir
+        {
+            get
+            {
+                if (Environment.CurrentDirectory.Contains("system32") && Environment.CurrentDirectory.Contains("Windows"))
+                    return null;
+                else
+                    return Path.Combine(Environment.CurrentDirectory, "QMods");
+            }
+        }
+
+        private static bool Patched = false;
+        internal static QModGame CurrentlyRunningGame { get; private set; } = QModGame.None;
+        internal static int ErrorModCount { get; private set; }
+
+        internal static void Patch()
+        {
+            try
+            {
+                if (Patched)
+                {
+                    Logger.Warn("Patch method was called multiple times!");
+                    return; // Halt patching
+                }
+
+                Patched = true;
+
+                Logger.Info($"Loading QModManager v{Assembly.GetExecutingAssembly().GetName().Version.ToStringParsed()}...");
+
+                if (QModBaseDir == null)
+                {
+                    Logger.Fatal("A fatal error has occurred.");
+                    Logger.Fatal("There was an error with the QMods directory");
+                    Logger.Fatal("Please make sure that you ran Subnautica from Steam/Epic/Discord, and not from the executable file!");
+
+                    Dialog.Show("A fatal error has occurred. QModManager could not be initialized.", Dialog.Button.SeeLog, Dialog.Button.Disabled, false);
```

Comment:
In this case we don't want to prevent the game from running, since the user can just run it without QMods, so having the second button `Dialog.Button.Close` would make more sense.